tsl/src/remote/connection.c

```diff
@@ -933,11 +951,23 @@ remote_connection_get_result(const TSConnection *conn)
 
 		if (busy == 1)
 		{
-			/* Busy, wait for readable */
+			uint32 events;
 			WaitEvent event;
+			long timeout_ms;
+			int ret;
 
-			ModifyWaitEvent(conn->wes, conn->sockeventpos, WL_SOCKET_READABLE, NULL);
-			WaitEventSetWait(conn->wes, -1, &event, 1, PG_WAIT_EXTENSION);
+			events = WL_SOCKET_READABLE;
+			if (endtime != TS_NO_TIMEOUT)
+				events |= WL_TIMEOUT;
```

Comment:
For simplicity, I'd specify `WL_TIMEOUT` always, no harm in it.

Reply:
it requires timeout >= 0